src/main/java/com/ably/kafka/connect/utils/StructToJsonConverter.java

```diff
@@ -79,13 +81,16 @@ private static Map<String, Object> structJsonMap(final Struct struct) {
                 case BOOLEAN:
                     structMap.put(field.name(), struct.getBoolean(field.name()));
                     break;
-                case STRING:
-                    structMap.put(field.name(), struct.getString(field.name()));
-                    break;
                 case BYTES:
-                    throw new ConnectException("Bytes are currently not supported for conversion to JSON.");
+                    final Object logicalValue = LogicalTypeConversions.tryGetLogicalValue(schemaType, struct.get(field.name()));
+                    if (!(logicalValue instanceof BigDecimal)) {
+                        throw new ConnectException("Bytes are currently not supported for conversion to JSON.");
```

Comment:
`ConnectionException` doesn't feel like a very descriptive exception type for this error. Do we have nothing more appropriate?

Reply:
This is the generic `ConnectException` that we use for connector exceptions, but yes I think we could create/use something a bit more descriptive and also an exception that belongs to our package. I'm thinking about something like `DataFormatException` 🤔 WDYT?

Reply:
I went with `UnsupportedJsonTypeException` 86c229d2061c2c189fb4e430591c1086c1f66fde

Reply:
Ah thanks, I didn't realise `Connect` here is actually the proper noun / brand name `Connect` as opposed to describing a network connection problem! This change seems good to me in any case.